aten/src/ATen/ParallelCommon.cpp

```diff
@@ -103,6 +107,17 @@ int intraop_default_num_threads() {
   if (nthreads == 0) {
 #if defined(FBCODE_CAFFE2) && defined(__aarch64__)
     nthreads = 1;
+#elif defined(__aarch64__) && defined(__APPLE__)
+    // On Apple Silicon there are efficient and performance core
+    // Restrict parallel algorithms to performance cores by default
+    int32_t num_cores = -1;
+    size_t num_cores_len = sizeof(num_cores);
+    if (sysctlbyname("hw.perflevel0.physicalcpu", &num_cores, &num_cores_len, nullptr, 0) == 0) {
+      if (num_cores > 1) {
+        nthreads = num_cores;
+        return num_cores;
+      }
+    }
```

Comment:
Should we use the default as before here when this fails instead of 0?

Reply:
Good point, moved #ifdef to fallback to `TaskThreadPoolBase::defaultNumThreads()` if `sysctl` fails for some reason